public/service-worker.js

```diff
@@ -0,0 +1,30 @@
+function getEndpoint() {
+  return self.registration.pushManager.getSubscription()
+  .then(function(subscription) {
+    if (subscription) {
+      return subscription.endpoint;
+    }
+
+    throw new Error('User not subscribed');
+  });
+}
+
+// Register event listener for the 'push' event.
+self.addEventListener('push', function(event) {
+  // Keep the service worker alive until the notification is created.
+  event.waitUntil(
+    getEndpoint()
+    .then(function(endpoint) {
+      self.registration.showNotification('Find Your Immunization', {
+        body: event.data.text(),
```

Comment:
This is where we need to validate - once the BE is firing push notifications we can confirm the structure of the event data so we can serve the notification properly.